js/models/conversations.js

```diff
@@ -2115,29 +2122,34 @@
             source,
             'private'
           );
-          window.log.info(`Notify called on a different conversation. expected: ${this.id}. actual: ${conversation.id}`);
+          window.log.info(`Notify called on a different conversation.
+                           Expected: ${this.id}. Actual: ${conversation.id}`);
         } catch (e) {
-          return Promise.reject('Failed to fetch conversation');
+          return Promise.reject(new Error('Failed to fetch conversation'));
         }
       }
 
       const isTypeAccepted = type === 'accepted';
-      const title = isTypeAccepted ? 'friendRequestAcceptedNotificationTitle' : 'friendRequestNotificationTitle';
-      const message = isTypeAccepted ? 'friendRequestAcceptedNotificationMessage' : 'friendRequestNotificationMessage';
-
-      conversation.getNotificationIcon().then(iconUrl => {
-        window.log.info('Add notification for friend request updated', {
-          conversationId: conversation.idForLogging(),
-        });
-        Whisper.Notifications.add({
-          conversationId: conversation.id,
-          iconUrl,
-          isExpiringMessage: false,
-          message: i18n(message, conversation.getTitle()),
-          messageSentAt: Date.now(),
-          title: i18n(title),
-        });
-      }); 
+      const title = isTypeAccepted
+        ? 'friendRequestAcceptedNotificationTitle'
+        : 'friendRequestNotificationTitle';
+      const message = isTypeAccepted
+        ? 'friendRequestAcceptedNotificationMessage'
+        : 'friendRequestNotificationMessage';
+
+      const iconUrl = await conversation.getNotificationIcon();
+      window.log.info('Add notification for friend request updated', {
+        conversationId: conversation.idForLogging(),
+      });
+      Whisper.Notifications.add({
+        conversationId: conversation.id,
+        iconUrl,
+        isExpiringMessage: false,
+        message: i18n(message, conversation.getTitle()),
+        messageSentAt: Date.now(),
+        title: i18n(title),
+      });
+      return Promise.resolve();
```

Comment:
We can remove `return Promise.resolve()` as `async` functions automatically resolve when they finish executing

Reply:
The linter doesn't like there being no return value for async functions, can either put the flag to ignore this warning or just leave the redundant return

Reply:
Is the rue being violated the `consistent-return`? If so then changing `return Promise.reject(...)` to `throw ...` should fix it

Reply:
Aah yes I thought this wasn't the case but I was missing the inline return on the first line